### DISCUSSION TITLE
New Laravel 11 listener discovery and actions

### DISCUSSION THREAD

**Discussion by edalzell** [lorisleiva/laravel-actions#282] [General]
In Laravel 11, listeners are automatically discovered (https://laravel.com/docs/11.x/events#event-discovery). However I suspect most actions are not in the `Listeners` folder so won't be discovered and you can't aim the discovery at the `Actions` folder because it only looks for a `handle` method with an event typehinted. Many of my actions use the `asListener` method so that the `handle` method has something more generic (like a model or data class).

Hoping this kickstarts a discussion on how we might improve this for Laravel 11 and Laravel Actions.

Anyone have ideas?

**Comment by edalzell** [comment]
One thing I can thing of is a PR into Laravel where you can set the method that is looked for, so we could add `asListener`.

**Reply by edalzell** [reply]
Took a quick look, the PR wouldn't be that bad, to have a configurable method name that's searched for.

**Reply by jvkassi** [reply]
wow , that would be great again for DX.
Thanks for this package :+1:
